kmip/tests/unit/pie/test_client.py

```diff
@@ -951,6 +952,119 @@ def test_activate_on_operation_failure(self):
 
     @mock.patch('kmip.pie.client.KMIPProxy',
                 mock.MagicMock(spec_set=KMIPProxy))
+    def test_revoke(self):
+        """
+        Test that the client can revoke a secret.
+        """
+        revocation_reason = enums.RevocationReasonCode.KEY_COMPROMISE
+        uuid = 'aaaaaaaa-1111-2222-3333-ffffffffffff'
+        revocation_message = 'Key compromised!'
+        compromise_occurrence_date = 1
+
+        status = enums.ResultStatus.SUCCESS
+        result = results.OperationResult(contents.ResultStatus(status))
+
+        with ProxyKmipClient() as client:
+            client.proxy.revoke.return_value = result
+            result = client.revoke(
+                revocation_reason, uuid, revocation_message,
+                compromise_occurrence_date)
+            client.proxy.revoke.assert_called_with(
+                revocation_reason, uuid, revocation_message,
+                DateTime(compromise_occurrence_date,
+                         enums.Tags.COMPROMISE_OCCURRENCE_DATE))
+            self.assertEqual(None, result)
+
+    @mock.patch('kmip.pie.client.KMIPProxy',
+                mock.MagicMock(spec_set=KMIPProxy))
+    def test_revoke_on_invalid_inputs(self):
+        """
+        Test that a TypeError exception is raised when trying to revoke a
+        secret with invalid inputs.
+        """
+        revocation_reason = enums.RevocationReasonCode.KEY_COMPROMISE
+        revocation_reason_invalid = "key compromise"
+
+        uuid = 'aaaaaaaa-1111-2222-3333-ffffffffffff'
+        uuid_invalid = 123
+
+        revocation_message = 'Key compromised!'
+        revocation_message_invalid = 123
+
+        compromise_occurrence_date = 1
+        compromise_occurrence_date_invalid = '1'
+
+        args = [revocation_reason_invalid, uuid, revocation_message,
+                compromise_occurrence_date]
+        with ProxyKmipClient() as client:
+            self.assertRaises(TypeError, client.revoke, *args)
```

Comment:
Let's use `self.assertRaisesRegexp` here to make sure the right error is being generated for each invalid input.